### PR TITLE
Allow alert text, subject and spike query key to be nested fields

### PR DESCRIPTION
**Pull request by Qmando**
Use lookup_es_key instead of document.get

### CODE REVIEW

**Comment by y-trobinso** [comment]
lgtm. Add tests maybe?

File: elastalert/alerts.py [hunk 1]
```diff
@@ -34,7 +35,8 @@ def _add_custom_alert_text(self):
         alert_text = self.rule.get('alert_text', '')
         if 'alert_text_args' in self.rule:
             alert_text_args = self.rule.get('alert_text_args')
```
Comment (talyian) [inline elastalert/alerts.py]: not from this CR, but .get('alert_text_args', []) or otherwise the next lines will raise

Comment (Qmando) [inline elastalert/alerts.py]: This is in an if block to make sure it exists. Though someone could do alert_text_args: 5 or something dumb if it's not in the JSON schema

**Comment by talyian** [comment]
lg2m

**Pull request merged**